cobalt/persistent_storage/persistent_settings.cc

```diff
@@ -113,49 +113,46 @@ bool PersistentSettings::GetPersistentSettingAsBool(const std::string& key,
                                                     bool default_setting) {
   base::AutoLock auto_lock(pref_store_lock_);
   auto persistent_settings = pref_store_->GetValues();
-  const base::Value* result = persistent_settings->FindKey(key);
-  if (result && result->is_bool()) return result->GetBool();
-  return default_setting;
+  absl::optional<bool> result = persistent_settings.FindBool(key);
+  return result.value_or(default_setting);
 }
 
 int PersistentSettings::GetPersistentSettingAsInt(const std::string& key,
                                                   int default_setting) {
   base::AutoLock auto_lock(pref_store_lock_);
   auto persistent_settings = pref_store_->GetValues();
-  const base::Value* result = persistent_settings->FindKey(key);
-  if (result && result->is_int()) return result->GetInt();
-  return default_setting;
+  absl::optional<int> result = persistent_settings.FindInt(key);
+  return result.value_or(default_setting);
 }
 
 double PersistentSettings::GetPersistentSettingAsDouble(
     const std::string& key, double default_setting) {
   base::AutoLock auto_lock(pref_store_lock_);
   auto persistent_settings = pref_store_->GetValues();
-  const base::Value* result = persistent_settings->FindKey(key);
-  if (result && result->is_double()) {
-    return result->GetDouble();
-  }
-  return default_setting;
+  absl::optional<double> result = persistent_settings.FindDouble(key);
+  return result.value_or(default_setting);
 }
 
 std::string PersistentSettings::GetPersistentSettingAsString(
     const std::string& key, const std::string& default_setting) {
   base::AutoLock auto_lock(pref_store_lock_);
   auto persistent_settings = pref_store_->GetValues();
-  const base::Value* result = persistent_settings->FindKey(key);
-  if (result && result->is_string()) return result->GetString();
+  const std::string* result = persistent_settings.FindString(key);
+  if (result) return *result;
   return default_setting;
 }
 
 std::vector<base::Value> PersistentSettings::GetPersistentSettingAsList(
     const std::string& key) {
   base::AutoLock auto_lock(pref_store_lock_);
   auto persistent_settings = pref_store_->GetValues();
-  base::Value* result = persistent_settings.Find(key);
-  if (result && result->is_list()) {
-#ifndef USE_HACKY_COBALT_CHANGES
-    return std::move(result).TakeList();
-#endif
+  const base::Value::List* result = persistent_settings.FindList(key);
+  if (result) {
+    std::vector<base::Value> values;
```

Comment:
Declare this outside the conditional scope so we can just `return values` once at the bottom instead of having a separate default return statement